### PR TITLE
util: Change the name of ListInDisk

### LINKED ISSUE #47733: Spill Improvement
## Enhancement

- [x] Change the name of `ListInDisk` and update comment
  https://github.com/pingcap/tidb/pull/47777
- [x] Add `DataInDiskByChunks` and replace the `DataInDiskByRows` in agg executor with `DataInDiskByChunks`
  https://github.com/pingcap/tidb/pull/48160
- [x] Support spill to disk for parallel hash aggregation
  - [x] support serialization and deserialization of aggregate function for spill
    https://github.com/pingcap/tidb/pull/46632
  - [x] support spill to disk for parallel hash aggregation
    https://github.com/pingcap/tidb/pull/49074
- [x] Support spill to disk for sort executor
  - [x] Put sort files into one folder
    https://github.com/pingcap/tidb/pull/48351
  - [x] Put `SortExec` and `TopNExec` into different files.
    https://github.com/pingcap/tidb/pull/48569
  - [x] Move sort tests into `sortexec` package
    https://github.com/pingcap/tidb/pull/48636
  - [x] Refine sort spill
    https://github.com/pingcap/tidb/pull/48872
- [x] Support parallel sort spill
   https://github.com/pingcap/tidb/pull/50747
- [x] Support spill to disk for topn executor
   https://github.com/pingcap/tidb/pull/51809
  - [x] Refine topn
     https://github.com/pingcap/tidb/pull/51271

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: ref #47733 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/util/chunk/row_in_disk_test.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func BenchmarkListInDiskGetRow(b *testing.B) {
 }
 
 type listInDiskWriteDisk struct {
```
Comment (guo-shaoge) [inline pkg/util/chunk/row_in_disk_test.go:142]: maybe grep to change all `ListInDisk` to `DataInDiskByRows`

**Pull request merged**